proposals/proposal-Implicit-default-optional-parameters.md

```diff
@@ -0,0 +1,156 @@
+# Implicit Default Optional Parameters
+
+* [x] Proposed
+* [x] Prototype: [Complete](https://github.com/AdamSpeight2008/roslyn-AdamSpeight2008/tree/master_Feature_ImplicitDefaultOptionalParameter)
+* [ ] Implementation: [In Progress](https://github.com/AdamSpeight2008/roslyn-AdamSpeight2008/tree/PostDev15_Feature_ImplicitDefaultOptionalParameters)
+* [ ] Unit Tests
+* [ ] Code Diagnostic Tool. 
+* [ ] Specification: [Not Started](pr/1)
+
+## Summary
+[summary]: #summary
+Assume an implicit default value for optional parameter.
+
+## Motivation
+[motivation]: #motivation
+
+Simplifies the type of the more common usage case, that of using the default value of the associated type as the default to value of the parameter.
+Eg `Foo(Optional size As String = Nothing)` to `Foo(Optional size As String)`
+
+## Detailed design
+[design]: #detailed-design
+
+The grammar definition for an `Optional Parameter` is something similar to this.
+
+```
+OptionalParameter ::= "Optional" ParameterName Typing? DefaultToValue?
+           Typing ::= "As" TypeIdentifier
+   DefaultToValue ::= "=" ( "Nothing" | ConstantValue )
+```
+
+**Examples**
+
+```vb.net
+Foo( Optional arg0 )                     ' --> Foo( Optional arg0 As Object = Nothing )
+Foo( Optional arg1 As String )           ' --> Foo( Optional arg1 As String = Nothing )
+Foo( Optional arg2 As String = Nothing ) ' --> Foo( Optional arg2 As String = Nothing )
+Foo( Optional arg3 = "" )                ' --> Foo( Optional arg3 As String = "" )
+Foo( Optional arg4 As Integer )          ' --> Foo( Optional arg4 As Integer = Nothing )
+Foo( Optional arg5 As Integer = 0 )      ' --> Foo( Optional arg5 As Integer = 0 )
+Foo( Optional arg6 As Integer = 1 )      ' --> Foo( Optional arg6 As Integer = 1 )
+Foo( Optional arg7 = 7 )                 ' --> Foo( Optional arg7 As Integer = 7 )
```

Comment:
This is a breaking change. I believe today this means Optional arg7 As Object = 7.